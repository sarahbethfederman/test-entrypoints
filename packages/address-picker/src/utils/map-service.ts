export default class MapService {
  private placesService: google.maps.places.PlacesService = {} as google.maps.places.PlacesService;
  private autocompleteService: google.maps.places.AutocompleteService = new google.maps.places.AutocompleteService();
  private sessionToken: google.maps.places.AutocompleteSessionToken = new google.maps.places.AutocompleteSessionToken();

  constructor(map?: google.maps.Map) {
    this.placesService = new google.maps.places.PlacesService(map || document.createElement('div'));
  }

  /**
   * Map enum to string
   * @param status google.maps.places.PlacesServiceStatus
   */
  private apiRejectionReason(status: google.maps.places.PlacesServiceStatus): Error {
    if (status === google.maps.places.PlacesServiceStatus.ZERO_RESULTS) {
      return new Error('ZERO_RESULTS');
    } else if (status === google.maps.places.PlacesServiceStatus.INVALID_REQUEST) {
      return new Error('INVALID_REQUEST');
    } else if (status === google.maps.places.PlacesServiceStatus.OVER_QUERY_LIMIT) {
      return new Error('OVER_QUERY_LIMIT');
    } else if (status === google.maps.places.PlacesServiceStatus.NOT_FOUND) {
      return new Error('NOT_FOUND');
    } else if (status === google.maps.places.PlacesServiceStatus.REQUEST_DENIED) {
      return new Error('REQUEST_DENIED');
    } else if (status === google.maps.places.PlacesServiceStatus.UNKNOWN_ERROR) {
      return new Error('UNKNOWN_ERROR');
    }
    return new Error('UNKNOWN_ERROR');
  }

  public placePredictions(
    input: string,
    componentRestrictions: google.maps.places.ComponentRestrictions
  ): Promise<google.maps.places.AutocompletePrediction[]> {
    return new Promise((resolve, reject) => {
      (this.autocompleteService as google.maps.places.AutocompleteService).getPlacePredictions(
        {
          input,
          sessionToken: this.sessionToken,
          componentRestrictions,
        },
        (suggestions: google.maps.places.AutocompletePrediction[], status: google.maps.places.PlacesServiceStatus) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && suggestions !== null) {
            resolve(suggestions);
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }

  public placeDetails(
    placeId: string
  ): Promise<{
    geometry: google.maps.places.PlaceGeometry | undefined;
    addressComponents: google.maps.GeocoderAddressComponent[];
  }> {
    return new Promise((resolve, reject) => {
      (this.placesService as google.maps.places.PlacesService).getDetails(
        {
          placeId,
          sessionToken: this.sessionToken,
        },
        (
          { geometry, address_components = [] }: google.maps.places.PlaceResult,
          status: google.maps.places.PlacesServiceStatus
        ) => {
          if (status === google.maps.places.PlacesServiceStatus.OK && address_components.length > 0) {
            resolve({
              geometry,
              addressComponents: address_components,
            });
          }
          reject(this.apiRejectionReason(status));
        }
      );
    });
  }
}
