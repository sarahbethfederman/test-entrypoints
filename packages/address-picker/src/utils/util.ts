import { AddressObject } from '../components/Modal';
import { STREET_TYPE } from '../components/Modal/constants';
import * as to from 'to-case';

/**
 * Where possible this function takes the users input and
 * Google's address information and enhances it with more
 * accurate subpremise/unit information
 */
export const transformGoogleResponse = (
  label: string,
  res: google.maps.GeocoderAddressComponent[],
  regionSearchOnly?: boolean
): AddressObject => {
  label = label.toLowerCase();

  // pick the easily determined properties from response
  const { short_name: postcode = '' } = res.find((curr) => curr.types.includes('postal_code')) || {};
  let { short_name: state = '' } = res.find((curr) => curr.types.includes('administrative_area_level_1')) || {};
  let { long_name: streetName = '' } = res.find((curr) => curr.types.includes('route')) || {};
  let { short_name: streetNumber = '' } = res.find((curr) => curr.types.includes('street_number')) || {};
  const { long_name: suburb = '' } = res.find((curr) => curr.types.includes('locality')) || {};
  const { long_name: country = '' } = res.find((curr) => curr.types.includes('country')) || {};
  let unit = '';

  // Google has no concept of "street". See if we recognise a streetType from
  // the route then use that to generate streetName and streetType
  // streetType default to 'street' if regionSearchOnly is false
  const street = streetName.trim().split(' ');
  const streetTypeWith = street[street.length - 1];
  let streetType = !regionSearchOnly ? 'Street' : '';
  if (STREET_TYPE.find((street_type) => street_type.label === streetTypeWith)) {
    streetType = streetTypeWith;
  }

  streetName = streetName.replace(streetType, '').trim();

  // edge case, state is listed as JBT for jervis bay when it should be ACT. Naughty Google
  if (state === 'JBT') {
    state = 'ACT';
  }

  // check for unit / subpremise info by crossreferencing against user input
  const streetNumberData = label.substring(0, label.indexOf(streetName.toLowerCase())).trim();
  if (streetNumberData !== streetNumber) {
    if (streetNumberData.includes('/')) {
      unit = streetNumberData.substring(0, streetNumberData.indexOf('/'));
    } else {
      streetNumber = streetNumberData;
    }
  }

  return {
    unit,
    lotSection: '',
    level: '',
    buildingName: '',
    streetNumber,
    streetName,
    streetType,
    suburb,
    postcode,
    state,
    country,
  };
};

export const getFormatedString = (address: AddressObject) => {
  let formatString = address.unit ? ''.concat(address.unit, '/') : '';
  formatString = address.lotSection ? formatString.concat('Lot. ', address.lotSection) : formatString;
  formatString = address.level ? formatString.concat(' level ', address.level) : formatString;
  formatString = address.buildingName ? formatString.concat(' ', to.capital(address.buildingName), ' ') : formatString;
  formatString = address.streetNumber ? formatString.concat(address.streetNumber) : formatString;
  formatString = address.streetName ? formatString.concat(' ', to.capital(address.streetName)) : formatString;
  formatString = address.streetType ? formatString.concat(' ', to.capital(address.streetType)) : formatString;
  formatString = address.suburb ? formatString.concat(', ', to.capital(address.suburb)) : formatString;
  formatString = address.state ? formatString.concat(', ', to.upper(address.state)) : formatString;
  formatString = address.postcode ? formatString.concat(' ', address.postcode) : formatString;
  formatString = address.country ? formatString.concat(', ', to.capital(address.country)) : formatString;
  return formatString;
};
