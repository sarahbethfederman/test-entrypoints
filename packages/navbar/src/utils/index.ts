export const browserSupportsPassiveListeners = () => {
  let supportsPassive = false;
  try {
    var opts = Object.defineProperty({}, 'passive', {
      get: function() {
        supportsPassive = true;
      },
    });
    // for testing browser compatibility, no need to worry
    // about the null TS issue.
    // @ts-ignore
    window.addEventListener('testPassive', null, opts);
    // @ts-ignore
    window.removeEventListener('testPassive', null, opts);
  } catch (e) {}

  return supportsPassive;
};
