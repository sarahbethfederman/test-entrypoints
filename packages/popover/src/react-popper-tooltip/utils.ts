import * as React from 'react';

export const TooltipContext = React.createContext({});

// eslint-disable-next-line @typescript-eslint/no-explicit-any
type Fn = (...args: any[]) => void;
type FnOrBool = undefined | Fn;

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const callAll = (...fns: FnOrBool[]) => (...args: any[]) => fns.forEach((fn) => fn && fn(...args));

export const noop = () => {};

export const canUseDOM = () => !!(typeof window !== 'undefined' && window.document && window.document.createElement);
