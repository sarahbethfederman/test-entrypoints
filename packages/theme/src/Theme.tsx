import * as React from 'react';
import styled, { ThemeProvider, injectGlobal } from 'styled-components';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
// tslint:disable-next-line no-unused-expression
injectGlobal`
  html, body {
    margin: 0;
    padding: 0;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: 36.0625em) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors.shade[700]};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>{children}</Root>
    </ThemeProvider>
  );
}
