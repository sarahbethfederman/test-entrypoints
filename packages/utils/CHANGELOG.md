# Changelog - @lendi-ui/utils

## [2.2.1] - 2019-05-08 - [HUB-54] (https://creditandfinance.atlassian.net/browse/HUB-54)
### Changed
- Change module main entry point to esm in package.json

## [2.2.0] - 2019-04-26 - [HUB-94](https://creditandfinance.atlassian.net/browse/HUB-94)
### Removed
- Created normalise mixin for LUI components.

## [2.1.0] - 2019-04-30 - [HUB-54] (https://creditandfinance.atlassian.net/browse/HUB-54)
### Added
- Added method to fetch all 'data-' props for a component.

## [2.0.0] - 2019-04-30 - [HUB-54] (https://creditandfinance.atlassian.net/browse/HUB-54)
### Deleted
- Remove the CJS bundles.

## [1.1.2] - 2019-03-08 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)
### Removed
- removed typescript type representing theme object.

## [1.0.1] - 2019-02-27 - [BRAVO-387](https://creditandfinance.atlassian.net/browse/BRAVO-387)
### Added
- typescript type representing theme object.