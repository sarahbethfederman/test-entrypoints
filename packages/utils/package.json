{
  "name": "@lendi-ui/utils",
  "version": "4.0.1",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "peerDependencies": {
    "@types/styled-components": "^4.1.18",
    "styled-components": "^4"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^4.0.0",
    "@lendi/lendi-analytics-package": "^1.0.0",
    "@types/lodash": "^4.14.144",
    "lodash": "^4.17.15",
    "prop-types": "^15.7.2"
  },
  "license": "ISC",
  "scripts": {
    "build:cjs": "tsc --project tsconfig.cjs.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:cjs"
  }
}
