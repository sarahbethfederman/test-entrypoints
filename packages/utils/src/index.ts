import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type DisplyValue = BreakpointValue<string> | BreakpointValueMap<string>;

export const display = (val: DisplyValue) => map(val, (v) => `display: ${v};`);
