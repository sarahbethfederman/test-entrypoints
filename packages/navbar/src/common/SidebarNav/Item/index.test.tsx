import * as React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import { Item } from './index';
import { ListItem, Link, Button } from './index.style';
import Theme from '@lendi-ui/theme';
import { AnalyticsContextProps } from '@lendi-ui/utils';
import { ItemProps } from '..';
import { WindowPosition } from '@lendi/lendi-analytics-package';

const mockAnalyticsContext: AnalyticsContextProps = {
  analyticsForNavigation: jest.fn(),
};

let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
function render(props: ItemProps) {
  wrapper = mount(
    <Theme>
      <Item {...props}>Foo Bar</Item>
    </Theme>
  );
  wrapper.find(Item).instance().context = mockAnalyticsContext;
  wrapper.update();
}

describe('Item', () => {
  describe('render a link', () => {
    beforeEach(() => {
      jest.resetModules();
      render({ href: '#foobar' });
    });
    it('when there is a href', () => {
      expect(wrapper.find(Link)).toHaveLength(1);
      expect(wrapper.find(Button)).toHaveLength(0);
      expect(wrapper.find(Link).props()).toEqual(
        expect.objectContaining({
          href: expect.any(String),
        })
      );
      expect(wrapper.find(Item)).toMatchSnapshot();
    });
  });

  describe('render a button', () => {
    beforeEach(() => {
      jest.resetModules();
      render({});
    });
    it('when there is not a href', () => {
      expect(wrapper.find(Link)).toHaveLength(0);
      expect(wrapper.find(Button)).toHaveLength(1);
      expect(wrapper.find(Button).props()).toEqual(
        expect.objectContaining({
          onClick: expect.any(Function),
        })
      );
      expect(wrapper.find(Item)).toMatchSnapshot();
    });
  });

  describe('click event', () => {
    beforeEach(() => {
      render({ href: '#foobar', labelText: 'Test item' });
    });
    it('should call AnalyticsContext fn', () => {
      expect(wrapper.find(Link)).toHaveLength(1);
      wrapper.find(Link).at(0).simulate('click');
      expect(mockAnalyticsContext.analyticsForNavigation).toBeCalledTimes(1);
    });
    it('should call AnalyticsContext fn with correct params', () => {
      expect(mockAnalyticsContext.analyticsForNavigation).toHaveBeenCalledWith(
        'Test item',
        WindowPosition.navigation_left
      );
    });
  });
});
