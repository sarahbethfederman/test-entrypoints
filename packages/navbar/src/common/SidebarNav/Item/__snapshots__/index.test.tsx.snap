// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item render a button when there is not a href 1`] = `
.c1 {
  height: 3rem;
  color: rgba(0,192,165,1);
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  padding-left: calc(1.5 * var(--lendi-ui-size));
  padding-right: calc(1.5 * var(--lendi-ui-size));
  font-weight: 600;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid rgba(227,227,227,1);
  padding-top: 0;
  padding-bottom: 0;
  border-top: none;
  border-right: none;
  border-left: none;
  background-color: transparent;
  width: 100%;
  outline: none;
}

.c1:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(0,192,165,1);
}

.c1:hover svg {
  stroke: rgba(255,255,255,1);
}

.c1:active {
  color: rgba(255,255,255,1);
  background-color: rgba(0,153,132,1);
}

.c1:last-type {
  border-bottom: none;
}

.c0 {
  height: 3rem;
  color: rgba(0,192,165,1);
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  padding-left: calc(1.5 * var(--lendi-ui-size));
  padding-right: calc(1.5 * var(--lendi-ui-size));
  font-weight: 600;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid rgba(227,227,227,1);
  padding: 0;
  list-style: none;
}

.c0:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(0,192,165,1);
}

.c0:hover svg {
  stroke: rgba(255,255,255,1);
}

.c0:active {
  color: rgba(255,255,255,1);
  background-color: rgba(0,153,132,1);
}

.c0:last-type {
  border-bottom: none;
}

.c0 > a {
  border-bottom: 0;
}

.c0 > button {
  border-bottom: 0;
}

<Item>
  <styled.li
    role="none"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              [Function],
              " ",
              "
    padding-left: ",
              "calc(1.5 * var(--lendi-ui-size))",
              ";
    padding-right: ",
              "calc(1.5 * var(--lendi-ui-size))",
              ";
  ",
              "
  font-weight: 600;
  cursor: pointer;
  display: flex;
  align-items: center;
  border-bottom: 1px solid ",
              [Function],
              ";

  :hover {
    ",
              "
          color: ",
              [Function],
              ";
        ",
              " ",
              "
          background-color: ",
              [Function],
              ";
        ",
              ";

    svg {
      stroke: ",
              [Function],
              ";
    }
  }

  :active {
    ",
              "
          color: ",
              [Function],
              ";
        ",
              " ",
              "
          background-color: ",
              [Function],
              ";
        ",
              ";
  }

  :last-type {
    border-bottom: none;
  }
",
              ";
  padding: 0;
  list-style: none;

  > a {
    border-bottom: 0;
  }

  > button {
    border-bottom: 0;
  }
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "li",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      role="none"
    >
      <li
        className="c0"
        role="none"
      >
        <styled.button
          depth={0}
          onClick={[Function]}
        >
          <StyledComponent
            depth={0}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htoDjs",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    " ",
                    "
    padding-left: ",
                    "calc(1.5 * var(--lendi-ui-size))",
                    ";
    padding-right: ",
                    "calc(1.5 * var(--lendi-ui-size))",
                    ";
  ",
                    "
  font-weight: 600;
  cursor: pointer;
  display: flex;
  align-items: center;
  border-bottom: 1px solid ",
                    [Function],
                    ";

  :hover {
    ",
                    "
          color: ",
                    [Function],
                    ";
        ",
                    " ",
                    "
          background-color: ",
                    [Function],
                    ";
        ",
                    ";

    svg {
      stroke: ",
                    [Function],
                    ";
    }
  }

  :active {
    ",
                    "
          color: ",
                    [Function],
                    ";
        ",
                    " ",
                    "
          background-color: ",
                    [Function],
                    ";
        ",
                    ";
  }

  :last-type {
    border-bottom: none;
  }
",
                    " ",
                    "
  ",
                    "
    padding-top: ",
                    "0",
                    ";
    padding-bottom: ",
                    "0",
                    ";
  ",
                    ";
  border-top: none;
  border-right: none;
  border-left: none;
  background-color: transparent;
  width: 100%;
  outline: none;
",
                    ";
",
                  ],
                },
                "displayName": "styled.button",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htoDjs",
                "target": "button",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            onClick={[Function]}
          >
            <button
              className="c1"
              onClick={[Function]}
            >
              Foo Bar
            </button>
          </StyledComponent>
        </styled.button>
      </li>
    </StyledComponent>
  </styled.li>
</Item>
`;

exports[`Item render a link when there is a href 1`] = `
.c1 {
  height: 3rem;
  color: rgba(0,192,165,1);
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  padding-left: calc(1.5 * var(--lendi-ui-size));
  padding-right: calc(1.5 * var(--lendi-ui-size));
  font-weight: 600;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid rgba(227,227,227,1);
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c1:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(0,192,165,1);
}

.c1:hover svg {
  stroke: rgba(255,255,255,1);
}

.c1:active {
  color: rgba(255,255,255,1);
  background-color: rgba(0,153,132,1);
}

.c1:last-type {
  border-bottom: none;
}

.c0 {
  height: 3rem;
  color: rgba(0,192,165,1);
  font-size: var(--lendi-ui-size);
  line-height: calc(24 / 16);
  font-family: 'Open Sans',sans-serif;
  padding-left: calc(1.5 * var(--lendi-ui-size));
  padding-right: calc(1.5 * var(--lendi-ui-size));
  font-weight: 600;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 1px solid rgba(227,227,227,1);
  padding: 0;
  list-style: none;
}

.c0:hover {
  color: rgba(255,255,255,1);
  background-color: rgba(0,192,165,1);
}

.c0:hover svg {
  stroke: rgba(255,255,255,1);
}

.c0:active {
  color: rgba(255,255,255,1);
  background-color: rgba(0,153,132,1);
}

.c0:last-type {
  border-bottom: none;
}

.c0 > a {
  border-bottom: 0;
}

.c0 > button {
  border-bottom: 0;
}

<Item
  href="#foobar"
>
  <styled.li
    role="none"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  ",
              [Function],
              " ",
              "
    padding-left: ",
              "calc(1.5 * var(--lendi-ui-size))",
              ";
    padding-right: ",
              "calc(1.5 * var(--lendi-ui-size))",
              ";
  ",
              "
  font-weight: 600;
  cursor: pointer;
  display: flex;
  align-items: center;
  border-bottom: 1px solid ",
              [Function],
              ";

  :hover {
    ",
              "
          color: ",
              [Function],
              ";
        ",
              " ",
              "
          background-color: ",
              [Function],
              ";
        ",
              ";

    svg {
      stroke: ",
              [Function],
              ";
    }
  }

  :active {
    ",
              "
          color: ",
              [Function],
              ";
        ",
              " ",
              "
          background-color: ",
              [Function],
              ";
        ",
              ";
  }

  :last-type {
    border-bottom: none;
  }
",
              ";
  padding: 0;
  list-style: none;

  > a {
    border-bottom: 0;
  }

  > button {
    border-bottom: 0;
  }
",
            ],
          },
          "displayName": "styled.li",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "li",
          "toString": [Function],
          "usesTheme": false,
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      role="none"
    >
      <li
        className="c0"
        role="none"
      >
        <styled.a
          depth={0}
          href="#foobar"
          onClick={[Function]}
          role="menu item"
        >
          <StyledComponent
            depth={0}
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  ",
                    "
  ",
                    [Function],
                    " ",
                    "
    padding-left: ",
                    "calc(1.5 * var(--lendi-ui-size))",
                    ";
    padding-right: ",
                    "calc(1.5 * var(--lendi-ui-size))",
                    ";
  ",
                    "
  font-weight: 600;
  cursor: pointer;
  display: flex;
  align-items: center;
  border-bottom: 1px solid ",
                    [Function],
                    ";

  :hover {
    ",
                    "
          color: ",
                    [Function],
                    ";
        ",
                    " ",
                    "
          background-color: ",
                    [Function],
                    ";
        ",
                    ";

    svg {
      stroke: ",
                    [Function],
                    ";
    }
  }

  :active {
    ",
                    "
          color: ",
                    [Function],
                    ";
        ",
                    " ",
                    "
          background-color: ",
                    [Function],
                    ";
        ",
                    ";
  }

  :last-type {
    border-bottom: none;
  }
",
                    ";
  text-decoration: none;
  width: 100%;
",
                  ],
                },
                "displayName": "styled.a",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "a",
                "toString": [Function],
                "usesTheme": false,
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            href="#foobar"
            onClick={[Function]}
            role="menu item"
          >
            <a
              className="c1"
              href="#foobar"
              onClick={[Function]}
              role="menu item"
            >
              Foo Bar
            </a>
          </StyledComponent>
        </styled.a>
      </li>
    </StyledComponent>
  </styled.li>
</Item>
`;
